core/src/org/labkey/core/query/DataStatesTableInfo.java

```diff
@@ -80,9 +91,36 @@ private boolean validateQCStateNotInUse(Map<String, Object> oldRowMap, Container
             return true;
         }
 
+        private String validateQCStateChangeAllowed(Map<String, Object> row, Container container)
+        {
+            Map<String, DataStateHandler> registeredHandlers = DataStateManager.getInstance().getRegisteredDataHandlers();
+            DataState qcChanging = DataStateManager.getInstance().getStateForRowId(container, (Integer) row.get("rowid"));
+
+            for (DataStateHandler handler : registeredHandlers.values())
+            {
+                String errorMsg = handler.isStateChangeAllowed(container, qcChanging, row);
+                if (errorMsg != null)
+                    return errorMsg;
+            }
+            return null;
+        }
+
+        private boolean validateLabel(Map<String, Object> row)
+        {
+            String label = (String) row.get("label");
+            return !StringUtils.isBlank(label);
+        }
+
         @Override
         protected Map<String, Object> updateRow(User user, Container container, Map<String, Object> row, @NotNull Map<String, Object> oldRow, boolean allowOwner, boolean retainCreation) throws InvalidKeyException, ValidationException, QueryUpdateServiceException, SQLException
         {
+            if (!validateLabel(row))
+                throw new QueryUpdateServiceException("State label cannot be blank.");
```

Comment:
```suggestion
                throw new QueryUpdateServiceException("Label cannot be blank.");
```
This avoids using "State" when the user sees nothing but "status".

Reply:
Done.